src/components/posts/AddPost.tsx

```diff
@@ -0,0 +1,141 @@
+import { joiResolver } from "@hookform/resolvers/joi";
+import axios from "axios";
+import { baseURL } from "@config";
+import { PostActionsEnums } from "@enums";
+import { usePost } from "@hooks";
+import { PostsProps } from "@interfaces";
+import { useEffect, ReactElement } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate, useParams } from "react-router-dom";
+import { PosterValidatorSchema } from "@validators";
+
+export const AddPost = (): ReactElement => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<PostsProps>({
+    mode: "onChange",
+    resolver: joiResolver(PosterValidatorSchema()),
+  });
+  const navigate = useNavigate();
+
+  const { createPost, updatePost } = usePost();
+  const { modal, id } = useParams();
+
+  const { CREATE, EDIT } = PostActionsEnums;
+
+  const handleCreate = () => {
+    navigate(`/posts/${CREATE}`);
+  };
+  const handleGoBack = () => navigate("/posts");
+
+  const onCreate = (data: PostsProps) => {
+    if (modal === CREATE) {
+      createPost(data);
+
+      handleGoBack();
+      reset();
+    } else {
```

Comment:
You can avoid the else here and just use one if condition, it's the same. But if you're going to add delete here aswel, I'd still recommend using just if conditions.